### PR TITLE
ref: first pass at removing some dead code in tests

### PR DESCRIPTION
**Pull request by asottile-sentry**
I only got through about ~30% of the test files -- will follow up with other ones

some explanation for some patterns:
- deleted function => nothing called it
- `raise SomeType` -> `raise AssertionError` -- `AssertionError` is excluded from coverage
- `pass` -> `raise NotImplementedError` -- these functions are never called -- only used for their shape / name and `NotImplementedError` is excluded from coverage
- `try: some_test_code except: fail(...)` -> no need for the `try`, an uncaught exception already fails the test

### CODE REVIEW

**Review by armenzg — APPROVED** [review]

File: tests/snuba/search/test_backend.py [hunk 1]
```diff
@@ -2567,25 +2566,18 @@ class EventsSnubaSearchTest(TestCase, EventsSnubaSearchTestCases):
...
-                resp = snuba._snuba_pool.urlopen(
-                    "POST",
-                    "/tests/entities/group_attributes/insert",
-                    body=json.dumps([snapshot]),
-                    headers={},
-                )
-                if resp.status != 200:
-                    raise snuba.SnubaError(
-                        f"HTTP {resp.status} response from Snuba! {json.loads(resp.data)}"
-                    )
```
Comment (wedamija) [inline tests/snuba/search/test_backend.py:2583]: Not totally sure I understand this change. It seems like we're losing out on a detailed message here

Comment (asottile-sentry) [reply tests/snuba/search/test_backend.py:2583]: even if it were useful it would never be hit -- urllib3 raise_for_status already happened

Comment (wedamija) [reply tests/snuba/search/test_backend.py:2583]: Is `raise_for_status` called inside of `urlopen`? I don't see it being called otherwise here.

Comment (asottile-sentry) [reply tests/snuba/search/test_backend.py:2583]: yeah

**Pull request merged**